conjure-go-client/httpclient/client_params.go

```diff
@@ -71,44 +72,33 @@ func (f clientOrHTTPClientParamFunc) applyHTTPClient(b *httpClientBuilder) error
 	return f(b)
 }
 
-func WithConfig(c ClientConfig) ClientParam {
-	return clientParamFunc(func(b *clientBuilder) error {
-		params, err := configToParams(c)
-		if err != nil {
-			return err
-		}
-		for _, p := range params {
-			if err := p.apply(b); err != nil {
-				return err
-			}
-		}
+func configOverrideClientParamFunc(override func(c *ClientConfig)) clientOrHTTPClientParamFunc {
```

Comment:
Can you add some documentation to explain what this is and when it should be used?